models/users.js

```diff
@@ -42,9 +42,17 @@ const addOrUpdate = async (userData, userId = null) => {
       const isNewUser = !user.data();
       // user exists update user
       if (user.data()) {
+        // remove id field if exist in fetched data or profileDiff
+        const currentData = user.data();
+        if ("id" in currentData) {
+          delete currentData.id;
+        }
+        if ("id" in userData) {
+          delete userData.id;
+        }
```

Comment:
Help me understand, why is this being done?

Reply:
In the current implementaton, user.data() is the present data of user being fetched and userData are the profileDiffs. In this section, I am checking if an id field in present in document data and then deleting it. As we need to filter out id field while pushing diffs.

Note: In line 59, we are using { merge : true } so we actually do not really need to fetch and push user.data() at all or work on currentData but it is there in the present code so i updated it accordingly. Discussed with lakshay to make a seperate issue for removal of same.